pkg/sriov/sriov.go

```diff
@@ -192,7 +197,12 @@ func (s *sriovManager) SetupVF(conf *sriovtypes.NetConf, podifName string, cid s
 			return fmt.Errorf("error setting container interface name %s for %s", linkName, tempName)
 		}
 
-		// 6. Bring IF up in Pod netns
+		// 6. Enable IPv4 ARP notify and IPv6 Network Discovery notify
+		if err := s.utils.enableArpAndNdiscNotify(podifName); err != nil {
+			return fmt.Errorf("failed to enable arp_notify for interface name: %s, %q", podifName, err)
```

Comment:
should we fail here ?

maybe it should be more like:

`tryEnableArpAndNdiscNotify()`

Reply:
Is there a concern that this function would fail in different environments? Or is there a scenario that you are thinking that would fail, but only fails because the sysfs variables aren't supported?

Reply:
I could only think of 2 scenarios where calling this function would fail:
1) Not enough permissions to set the sysctl variable
2) The podifName is incorrect.
However both of these scenarios would be considered major malfunctions and would require us to log an error.
@adrianchiris

Reply:
so, my question is, whether we regard this as an optimization or a mandatory operation.

for the former we should not fail. for the latter we should.

Reply:
according to this PR cover letter:

```
For performance reasons,
sending of GARP and/or Unsolicited NA would update invalid ARP/Neighbor
caches in other neighbors/nodes.
```

its the former

Reply:
@adrianchiris 

I see what you mean. There are 2 roadblocks I have:
1) SRIOV-CNI is a Multus plugin and thus doesn't have the ability to log warnings/informational. We can only report errors, but this in turn causes the pods to fail to create.
2) Yes, this is an optimization. But the causes for this function to fail would be abnormal. There would be no reason for a healthy system to fail calling these functions.

Are we ok with running the function and if it fails, we don't log anything? Otherwise we need a new logging feature for SRIOV-CNI.

Reply:
Under the premise that this is strictly an optimization (i.e. there are no other hard-to-debug side-effects of not having this in place), I don't mind not logging anything.

Reply:
@adrianchiris 

In the spirit of moving this PR closer to being merged. I agree that it is an optimization, and there shouldn't be any hard to debug side effects. I am ok with removing any logging and just throwing away the error message. Is this what you had in mind?

Reply:
I made the change to not fail with error.